### PR TITLE
Support std and sgx

### PR DESCRIPTION
**Pull request by echevrier**
- Add changes from https://github.com/mesalock-linux/http_req-sgx to support sgx
- support only rust-tls

### CODE REVIEW

**Comment by echevrier** [comment]
The code compiled locally and work with the teeracle.

File: src/lib.rs [hunk 1]
```diff
@@ -13,6 +13,25 @@
 //!    println!("Status: {} {}", res.status_code(), res.reason());
 //!}
 //!```
+#![cfg_attr(all(feature = "sgx", not(target_env = "sgx")), no_std)]
+#![cfg_attr(target_env = "sgx", feature(rustc_private))]
```
Comment (murerfel) [inline src/lib.rs]: I'm not sure if we need these, in our own libraries, we just have:
`#![cfg_attr(not(feature = "std"), no_std)]` 

But if it works with these attributes defined, then we can leave them I think.

File: Cargo.toml [hunk 2]
```diff
@@ -10,25 +10,44 @@ categories = ["web-programming::http-client", "network-programming"]
...
+sgx_tstd = { version = "1.1.4", git = "https://github.com/haerdib/incubator-teaclave-sgx-sdk", branch = "master",optional = true ,features = ["net", "untrusted_time"]}
 
-[features]
-default = ["native-tls"]
-rust-tls = ["rustls", "webpki", "webpki-roots"]
+# sgx enabled external libraries
+unicase_sgx      = { package = "unicase",        git = "https://github.com/mesalock-linux/unicase-sgx", optional = true}
+rustls_sgx       = { package = "rustls",         git = "https://github.com/mesalock-linux/rustls",                   branch = "mesalock_sgx",        optional = true }
+webpki_sgx       = { package = "webpki",         git = "https://github.com/mesalock-linux/webpki",                   branch = "mesalock_sgx",        optional = true }
+webpki-roots_sgx = { package = "webpki-roots",   git = "https://github.com/mesalock-linux/webpki-roots",             branch = "mesalock_sgx",        optional = true }
```
Comment (murerfel) [inline Cargo.toml]: We don't tab-align the dependencies (anymore), it's too much effort and questionable benefit.

File: Cargo.toml [hunk 3]
```diff
@@ -10,25 +10,44 @@ categories = ["web-programming::http-client", "network-programming"]
...
 
+#[dependencies.native-tls]
+#version = "^0.2"
+#optional = true
+
 [dependencies]
-unicase = "^2.6"
+# sgx dependencies
+#sgx_tstd = { rev = "v1.1.3", git = "https://github.com/apache/teaclave-sgx-sdk.git", optional = true ,features = ["net", "untrusted_time"]}
+sgx_tstd = { version = "1.1.4", git = "https://github.com/haerdib/incubator-teaclave-sgx-sdk", branch = "master",optional = true ,features = ["net", "untrusted_time"]}
```
Comment (murerfel) [inline Cargo.toml]: The spacing is off here:
```suggestion
sgx_tstd = { version = "1.1.4", git = "https://github.com/haerdib/incubator-teaclave-sgx-sdk", branch = "master", optional = true, features = ["net", "untrusted_time"] }
```

**Review by murerfel — COMMENTED** [review]
Just some minor formatting issues, fine otherwise 👍

File: src/lib.rs (hunk 1 above)
Comment (echevrier) [reply src/lib.rs]: #![cfg_attr(all(feature = "sgx", not(target_env = "sgx")), no_std)] is required, else the teeracle enclave runtime doesn't compile: duplicate lang item in crate `sgx_tstd` (which `http_req` depends on)
#![cfg_attr(target_env = "sgx", feature(rustc_private))], I don't know.

**Review by murerfel — APPROVED** [review]
LGTM 👍

**Pull request merged**